src/Common/EventFD.cpp

```diff
@@ -0,0 +1,63 @@
+
+#if defined(OS_LINUX)
+
+#include <Common/EventFD.h>
+#include <Common/Exception.h>
+#include <sys/eventfd.h>
+#include <unistd.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int CANNOT_PIPE;
+    extern const int CANNOT_READ_FROM_SOCKET;
+    extern const int CANNOT_WRITE_TO_SOCKET;
+}
+
+EventFD::EventFD()
+{
+    fd = eventfd(0, 0);
```

Comment:
```suggestion
    fd = eventfd(0 /* initval */, 0 /* flags */);
```

btw, shouldn't we set `O_NONBLOCK`?

Reply:
In this case no.
(We specially want to block till thread is finished, e.g. in ~AsyncReadingState).
For EventFD we can support both flags.
But I decided to delay till needed.

Reply:
let's maybe state this in comment, 'cause it is not obvious